CMakeLists.txt

```diff
@@ -219,12 +220,6 @@ endif()
 # VFX 18 deprecations to transition to MINIMUM_* variables in OpenVDB 7.0.0
 
 set(FUTURE_MINIMUM_CMAKE_VERSION 3.12)
-set(FUTURE_MINIMUM_GCC_VERSION 6.3.1)
-set(FUTURE_MINIMUM_ICC_VERSION 17)
-set(FUTURE_MINIMUM_MSVC_VERSION 19.10)
-set(FUTURE_MINIMUM_TBB_VERSION 2017.6)
-set(FUTURE_MINIMUM_NUMPY_VERSION 1.12.1)
-set(FUTURE_MINIMUM_HOUDINI_VERSION 17)
```

Comment:
These will cause problems if they don't exist when they're being checked - and it seems like this PR only comments out the Houdini ones? Maybe it's better to leave these in but set them to their above counterparts

Reply:
Perhaps it's best to have a global flag that disables them which we only switch on one or two minor releases before we finally remove support? I understand VERSION<MINIMUM to mean "no longer supported" and VERSION<FUTURE_MINIMUM to mean "deprecated but still supported".

If that's the case, this would mean that Houdini 17.0 and Houdini 17.5 are "deprecated" which is a hard argument to make considering Houdini 18.0 has yet to be released.

Reply:
Ok, so I've added an OPENVDB_FUTURE_DEPRECATION flag which currently defaults to OFF. If you switch it on, you can see which third-party libraries/applications you're using that are due to be deprecated in the future. I figure that we can switch the default value of this flag as we get closer to the next major release. I've also guarded by these variables existing as they won't always be in use.

Reply:
Nice work :+1: